File: Task/Day-of-the-week-of-Christmas-and-New-Year/Python/day-of-the-week-of-christmas-and-new-year.py
import datetime

weekDays = ("Monday","Tuesday","Wednesday","Thursday","Friday","Saturday","Sunday")
thisXMas  = datetime.date(2021,12,25)
thisXMasDay = thisXMas.weekday()
thisXMasDayAsString = weekDays[thisXMasDay]
print("This year's Christmas is on a {}".format(thisXMasDayAsString))

nextNewYear = datetime.date(2022,1,1)
nextNewYearDay = nextNewYear.weekday()
nextNewYearDayAsString = weekDays[nextNewYearDay]
print("Next new year is on a {}".format(nextNewYearDayAsString))

File: Task/Day-of-the-week-of-Christmas-and-New-Year/Ruby/day-of-the-week-of-christmas-and-new-year.rb
require 'date'

years = [1578, 1590, 1642, 1957, 2020, 2021, 2022, 2242, 2245, 2393]
years.each do |year|
  xmas = Date.new(year,12,25).strftime("%A")
  ny =   Date.new(year, 1, 1).strftime("%A")
  puts "In #{year}, New year's day is on a #{ny}, and Christmas day on #{xmas}."
end
